### DISCUSSION TITLE
DBeaver blocked by Windows Firewall after install, not sure what it's trying to do

### DISCUSSION THREAD

**Discussion by gearskullguy** [dbeaver/dbeaver#12038] [Q&A]
#### System information: 
- Operating system: Windows 10 Pro 64-bit Version 20H2
- DBeaver version: Community Version 21.0.1.202103211229
- Java version: None installed... it must be using a version that came with it.
- Additional extensions: ? I just installed this for the first time on my computer fresh from your website.

#### Connection specification:
- Database name and version: I just opened the app; got the firewall warning.
- Driver name: ?
- Do you use tunnels or proxies (SSH, SOCKS, etc)? N/A

#### Describe the problem you're observing:

I don't see this as being reported yet, so I'm reporting it here, because I'm letting the firewall block whatever dbeaver.exe is trying to do, because I don't know what it's trying to do yet. I don't need anything on my network or the internet accessing my install of DBeaver... that I know of.

![image](https://user-images.githubusercontent.com/32692076/112544491-afcf6d80-8d7c-11eb-96d3-eb54215c3659.png)

Also why does Publisher say Unknown? A publisher was listed during install.

#### Steps to reproduce, if exist:

1. Install DBeaver from the official site, onto Windows 10.
2. Open it up for the first time.
3. Be miffed at a Windows Firewall prompt saying it's blocked something with dbeaver.exe.
4. Be baffled at deciding whether to leave Private Networks checked, or do anything in the prompt, or just click Cancel.
5. Wonder what the executable was trying to do that your firewall thought was a security risk.

#### Include any warning/errors/backtraces from the logs

Error Log is empty.

**Comment by kseniiaguzeeva** [comment]
DBeaver connects to dbeaver.io to get info about version upgrade. You can disable version check in Window>Preferences>User interface
![Screenshot 2021-04-08 at 10 40 47](https://user-images.githubusercontent.com/51405061/113987313-ef0abd80-9856-11eb-865f-ce8b37b7a700.png)

**Comment by hazcod** [comment]
@kseniiaguzeeva are you sure? The windows firewall only prompts for INCOMING listeners on 0.0.0.0.
An update check should be an OUTBOUND connection which simply traverses over the firewall.
This sounds like something that DBeaver shouldn't listen on?

**Comment by ShadelessFox** [comment]
Hello @hazcod,

DBeaver handles CLI commands execution using sockets, so you can operate on an already running instance instead of launching a new one.

I believe this triggers firewall.

**Reply by hazcod** [reply]
But sockets should be on the local filesystem or be localhost too?
